cabby/rvs/generate_rvs.py

```diff
@@ -125,15 +128,29 @@ def main(argv):
       instructions=gen_instructions,
       id=entity_idx,
     )
-    gen_samples.append(rvs_entity.sample)
+    gen_samples.append(rvs_entity)
+
+  logging.info(f"RVS generated: {len(gen_samples)}")
+
+  unique_instructions = []
+  unique_gen_samples = []
+  for gen in gen_samples:
+    if gen.instructions not in unique_instructions:
```

Comment:
It's very expensive to check for presence of an item in a list. For this, you can use a Dict, along the following lines:

```
uniq_samples = {}
for gen in gen_samples:
  uniq_samples[gen.instructions] = gen
``` 

This will then mean that you get the last gen item corresponding to a set of duplicate instructions, and it uses fast dictionary assignment instead of linear search over a list. Then, you just do uniq_samples.values() to get the gen samples below.